qcodes/tests/test_visa.py

```diff
@@ -51,11 +52,9 @@ def write(self, cmd):
             raise ValueError('be more positive!')
 
         if num == 0:
-            ret_code = visa.constants.VI_ERROR_TMO
-        else:
-            ret_code = 0
+            raise visa.VisaIOError(visa.constants.VI_ERROR_TMO)
```

Comment:
The test `test_ask_write_local` was failing after removing the `ret_code` and `error_check` from `visa.py`. I don't think my solution to get the test to pass is actually testing how visa's resource handle is used. I wanted to get your thoughts @jenshnielsen and @astafan8 before I merge this PR.

Reply:
I think this is fine. It looks like the intention of this mock all the visa handle out and only test qcodes handling of the visa commands. The tests are not of particulary high quality but I don't think they are made any worse by doing this